src/Components/Layout/Admin/AdminLayout.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react'
+import { NavLink, Outlet } from 'react-router-dom'
+import { SidebarMenu } from '../LayoutComponents/SidebarMenu'
+import Header from '../LayoutComponents/Header'
+import { filterModalStyle } from '../../Style'
+import { style } from '../../CustomComponent/CustomModal'
+
+const AdminLayout = () => {
+  const [openSubMenus, setOpenSubMenus] = React.useState(new Array(SidebarMenu.length).fill(false))
+  const [open, setOpen] = React.useState(true)
+  const [selectedSubcategory, setSelectedSubcategory] = React.useState<any>(null)
+
+  const handleExpand = (index: number) => {
+    setOpenSubMenus((prev) => {
+      const updatedSubMenus = new Array(SidebarMenu.length).fill(false)
+      updatedSubMenus[index] = !prev[index]
+      return updatedSubMenus
+    })
+  }
+
+  React.useEffect(() => {
```

Comment:
Why just not import the useEffect and useState ?
is their any benefit of this way ?

Reply:
No Actually , Using React.useEffect makes it clear that we're using a hook from the React library, which can help with understanding the code. However, if you think importing useEffect directly would be better for our project goals, I'm open to doing that instead.

Reply:
even official doc https://react.dev/learn does not use this method of importing

Reply:
Actually, I didn't have a specific link for this. Honestly, I looked it up on the internet because this sidebar has multiple functionalities on its own. Because of that, this file had the import issue. But now, I've fixed it.